File: prisma/refs/java/decode-ways.java
class Solution {
    public int numDecodings(String s) {
        if (s.charAt(0) == '0') return 0;
        int prev = 1, cur = 1;
        for (int i = 1; i < s.length(); i++) {
            int ways = 0;
            if (s.charAt(i) != '0') ways += cur;
            int two = Integer.parseInt(s.substring(i - 1, i + 1));
            if (two >= 10 && two <= 26) ways += prev;
            prev = cur;
            cur = ways;
        }
        return cur;
    }
}

File: prisma/refs/py/decode-ways.py
def num_decodings(s):
    if s[0] == "0":
        return 0
    prev, cur = 1, 1
    for i in range(1, len(s)):
        ways = 0
        if s[i] != "0":
            ways += cur
        if 10 <= int(s[i - 1:i + 1]) <= 26:
            ways += prev
        prev, cur = cur, ways
    return cur
